### PR TITLE
Added polymorphic execution policy and corresponding unit tests

### PR DESCRIPTION
**Pull request by jdgarciauc3m**
The new polymorphic execution policy may hold any other policy.

### CODE REVIEW

File: include/common/polymorphic_execution.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+#ifndef GRPPI_POLYMORPHIC_EXECUTION_H
+#define GRPPI_POLYMORPHIC_EXECUTION_H
+
+#include <typeinfo>
+#include <memory>
+
+#include "common/seq_policy.h"
+#include "common/thread_policy.h"
+#include "common/tbb_policy.h"
```
Comment (Drioastorga) [inline include/common/polymorphic_execution.h:29]: This include should be protected with "GRPPI_TBB".

File: include/common/polymorphic_execution.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#include "common/omp_policy.h"
```
Comment (Drioastorga) [inline include/common/polymorphic_execution.h:30]: This include should be protected with "GRPPI_OMP".

File: include/common/polymorphic_execution.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+namespace grppi{
+
+/// Meta-function to determine if a type is an execution policy
+template <typename E>
+constexpr bool is_execution_policy() {
+  return std::is_same<E, grppi::sequential_execution>::value 
+      || std::is_same<E, grppi::parallel_execution_thr>::value
+      || std::is_same<E, grppi::parallel_execution_tbb>::value
```
Comment (Drioastorga) [inline include/common/polymorphic_execution.h]: This execution policy type will not exists if the system does not support a given framework. I propose two different solutions to this issue. The first solution is to wrap each type with the corresponding "ifdef". The second solution is modify the policies code to define empty classes for those frameworks that are not supported. This also apply to the OpenMP execution policy.

File: include/common/polymorphic_execution.h (hunk 1 above)
Comment (jdgarciauc3m) [reply include/common/polymorphic_execution.h:29]: This is done

**Review by Drioastorga — CHANGES_REQUESTED** [review]
The different execution policy includes should be correspondingly protected in order  to avoid potential dependencies that are not available in the system. Additionally,  there is a related issue when checking the type contained in the polymorphic policy. This checking should only consider the supported execution policies.

**Pull request merged**